pkg/k8s/state_into_spec_annotation.go

```diff
@@ -19,24 +19,19 @@ import (
 	"strings"
 
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
 	"k8s.io/apimachinery/pkg/runtime/schema"
 )
 
-func ValidateOrDefaultStateIntoSpecAnnotation(obj *unstructured.Unstructured) error {
+// ValidateOrDefaultStateIntoSpecAnnotation validates the value of the
+// 'state-into-spec' annotation if it is set and defaults the annotation to the
+// passed in defaultValue if it is unset.
+func ValidateOrDefaultStateIntoSpecAnnotation(obj metav1.Object, gvk schema.GroupVersionKind, defaultValue string) error {
 	_, found := GetAnnotation(StateIntoSpecAnnotation, obj)
 	if !found {
-		SetAnnotation(StateIntoSpecAnnotation, StateMergeIntoSpec, obj)
+		// TODO(b/322836859): Ensure ComputeAddress doesn't need special handling.
```

Comment:
I think we can drop this todo, because #1162 has merged?

Reply:
Actually b/322839663 is also necessary. But will move it anyways since https://github.com/GoogleCloudPlatform/k8s-config-connector/pull/1178 is out.